### DISCUSSION TITLE
Component handles offset from component in design view

### DISCUSSION THREAD

**Discussion by owlbrudder** [eclipse/windowbuilder#306] [Q&A]
Over a year ago I had the same problem with WindowBuilder and it was never resolved.

I run Fedora Linux - right now on Fedora 36. The problem is the same under Gnome  Wayland and Gnome X11.

I have a new computer - this one is AMD not Intel for what it is worth. A fresh install of Eclipse and WindowBuilder.

As you can see, when I drop a component into place, the component is offset from its handles.

If I restart Eclipse and look at the design window again, the components are correctly aligned with their handles but any attempt to move a component creates the offset problem again.

For what it is worth, I do not have the same problem dropping and moving report components in BIRT.

Here is my Eclipse configuration data obtained using CTRL-3:
[windowbuilder2.txt](https://github.com/eclipse/windowbuilder/files/9034197/windowbuilder2.txt)

Here is a sequence of screenshots showing the behaviour before and after restarting Eclipse. I added a text field, restarted Eclipse, added another text field and restarted again.
![Screenshot from 2022-07-03 14-17-34](https://user-images.githubusercontent.com/20263907/177028650-f5ab8346-d251-43e8-b5e5-9a9e757679f1.png)
![Screenshot from 2022-07-03 14-18-03](https://user-images.githubusercontent.com/20263907/177028659-516f50fe-6719-4064-b3c7-3dd04f63f7f4.png)
![Screenshot from 2022-07-03 14-18-57](https://user-images.githubusercontent.com/20263907/177028666-8d695666-716f-481b-98ea-2016799ddcf4.png)
![Screenshot from 2022-07-03 14-19-24](https://user-images.githubusercontent.com/20263907/177028669-57e83935-1850-4013-afd5-6c51b1cd5761.png)

**Comment by wimjongman** [comment]
Doug, I thought this was resolved by you. See: https://bugs.eclipse.org/bugs/show_bug.cgi?id=570994

**Reply by owlbrudder** [reply]
Yes Wim, the problem went away when I repaved the machine. That was a drastic cure and it did not find the source of the problem. Now on a brand new computer with a brand new installation of Eclipse, running a newer Fedora and the problem has returned.

I am going to try a couple of ideas, but at first glance I seem to be back in the same problem.

**Comment by owlbrudder** [comment]
I can now state what is NOT the cause:

The CPU architecture: Intel Core-i7 and Ryzen 5600G do the same
The amount of RAM: I now have 32Gb
The version of Fedora
The version of Eclipse
The version of WindowBuilder
The window manager: X11 and Wayland do the same

What I can try to do is show you a screenshot video demonstrating what happens when I move an existing component after restarting Eclipse. Everything starts in the right place, but moving the object triggers the problem. The fact that it starts OK means the problem is somewhere in the WindowBuilder code to do with moving? placing? an object.

[Screencast from 07-04-2022 08:59:33 PM.webm](https://user-images.githubusercontent.com/20263907/177161432-da3d7ca7-09e6-47d2-afd1-f32e18b4fcc4.webm)

**Comment by wimjongman** [comment]
I'm not on Linux. @vogella Lars, are you?

**Reply by owlbrudder** [reply]
Wim, may I ask what JDK you are using? Matching your working configuration is one thing I have not tried.

**Reply by wimjongman** [reply]
I am on windows. I work on whatever Java version comes with Eclipse these days. I don't install java separately.

**Reply by vogella** [reply]
I only use WB on Windows at the moment.

**Reply by owlbrudder** [reply]
I have spent the afternoon here installing a completely fresh instance of Eclipse with BIRT and WindowBuilder. Sounds easy if I say it quickly. I will give it all a workout tomorrow and let you know what I find.

**Reply by owlbrudder** [reply]
After three days trying every incantation in my book of spells, I am still unable to install WindowBuilder on my platform without the problem of the displaced wireframes.

I have left Eclipse as it is in the hope that a solution will be found, but in order to progress my projects I have resorted to using NetBeans, which does not exhibit the wireframe problem. I will use NetBeans for all windows and Eclipse for all reports as I plan to use BIRT.

Many thanks for all the help so far.
